Homework №2 from 16.09.2022/task2.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+
+double slowPow(double number, int power)
+{
+    if (power == 0)
+    {
+        return 1;
+    }
+    else
+    {
+        return slowPow(number, power - 1) * number;
+    }
+}
+
+double fastPow(double number, int power)
+{
+    if (power == 0)
+    {
+        return 1;
+    }
+    if (power % 2 == 1)
+    {
+        return fastPow(number, (power - 1) / 2) * fastPow(number, (power - 1) / 2) * number;

```

Comment:
Двойной вызов функции с одинаковыми параметрами. Нужно вычислить один раз, а результат перемножить.

Reply:
```suggestion
        return fastPow(number, power / 2) * fastPow(number, power / 2) * number;
```
Мы знаем, что число нечетное в этой ветке. Деление нацело гарантировано округлит в меньшую сторону.